### PR TITLE
fix(component-testing): remove flake in vite-dev-server when the dependency chain is long

### LINKED ISSUE #16069: ci: design-system tests are flakey
### Current behavior
Whan a new Pull request is run on cypress, once every two build the npm-design-system job fails on CircleCi. It makes the build process tedious and a significant loss of time.

### Desired behavior
Not to have failures of these tests anymore.

### Versions

cypress 7.1.0

### PR DESCRIPTION
**Pull request by elevatebart**
- Closes #16069

### User facing changelog
Impoved stability when testing using `@cypress/vite-dev-server`. 
Vite had a hard time compiling long dependency chains.

### Additional details
- Mostly visible in electron in run mode in first CI because none of the dependencies are already pre-compiled and optimized yet.

### CODE REVIEW

File: npm/vite-dev-server/client/initCypressTests.js [hunk 1]
```diff
@@ -6,10 +6,32 @@ const originAutUrl = import.meta.env.__cypress_originAutUrl
 
 const specPath = window.location.pathname.replace(originAutUrl, '')
 
-const importsToLoad = [() => import(/* @vite-ignore */ specPath)]
+/**
+ * It was necessary here to wrap the imports
+ * in setTimeout to avoid for the mports to run before
```
Comment (lmiller1990) [inline npm/vite-dev-server/client/initCypressTests.js:11]:
```suggestion
 * in setTimeout to avoid for the imports to run before
```

**Review by lmiller1990 — DISMISSED** [review]
Interesting, is 1ms always going to be enough? Is the problem solved by delaying to the next tick using `setTimeout` and `Promise`? Do we still have a potential race condition problem?

Also, how can I test this? Any way to see it failing before vs fixed after?

**Comment by elevatebart** [comment]
@lmiller1990 to test this you would have to copy this package in an independent repository (Or clone [this one](https://github.com/elevatebart/cy-ds-test)). Without the fix, the SearchObject tests fail consistently even locally.

File: npm/vite-dev-server/client/initCypressTests.js [hunk 2]
```diff
@@ -6,10 +6,32 @@ const originAutUrl = import.meta.env.__cypress_originAutUrl
...
+ * vite is connected and ready to deal with errors
+ *
+ * If not, the imports of new dependencies failures
+ * would kill tests before they start
+ */
+const importsToLoad = [() => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      import(/* @vite-ignore */ specPath).then(resolve)
+    }, 1)
```
Comment (agg23) [inline npm/vite-dev-server/client/initCypressTests.js:21]: Please change these to 0ms (next tick).

**Comment by elevatebart** [comment]
After running a lot of iterations on CLI there is still one flake out of 20ish so I will try and find a better solution.

**Pull request closed without merging**